libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.h

```diff
@@ -48,7 +48,8 @@
 #define HTTPCLIENT_DEFAULT_TCP_TIMEOUT (5000)
 
 /// HTTP client errors
-#define HTTPC_ERROR_CONNECTION_REFUSED  (-1)
+#define HTTPC_ERROR_CONNECTION_REFUSED  (-1) // backward compatibility
```

Comment:
The third answer ("You could make sure that these macros would expand to something that would include a expression that would trigger the attribute((deprecated)) warning...." ) here
https://stackoverflow.com/questions/2681259/how-to-deprecate-a-macro-in-gcc
shows a little convoluted way of making it work.  Did not try myself, though.

Also, might be possible to make a single `typedef __attribute__ (( deprecated )) enum {HTTPC_ERROR_CONNECTION_REFUESD=-1};` ?

Reply:
what about `constexpr int HTTPC_ERROR_CONNECTION_REFUSED = -1 __attribute__((deprecated));`
and push milestone to v3 (because of `deprecated`) ?

Reply:
Ah, was thinking C and not C++, but this is a class so you're not using it in C anyway.  D'oh!

Why not deprecate now?  If we can deprecate SPIFFS in 2.7.1, doing a single, poorly named constant in 2.7.2 seems relatively OK.